src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -807,6 +807,43 @@ private void AddPythonEnginesOptions()
         /// </summary>
         public TrustedPathViewModel TrustedPathsViewModel { get; set; }
 
+        /// <summary>
+        /// Import Settings
+        /// </summary>
+        /// <param name="filePath"></param>
+        public void importSettings(string filePath)
+        {
+            PreferenceSettings newPreferences = dynamoViewModel.ImportPreferences(filePath);
+            newPreferences.CopyProperties(preferenceSettings);
```

Comment:
Would you explain why some of the properties can be copied but others need to be set explicitly?

Reply:
Because they don't have implemented the set method in their properties, that means their Can't be Written, That's why I need to access to the fields using the explicit methods like SetTrustWarningsDisabled and SetTrustedLocations.

Reply:
maybe you could add internal setters, unless there was a reason we didn't add those @pinzart90 ?

Reply:
Well trusted locations was designed to have a very restrictive API in order to make it as secure as possible (i.e make it hard to be hacked)
In order for Trusted location to be set, we would first need to check if it is valid (exists, read access, redundant paths etc)
We do have a example of setting the trusted locations blindly (without checks) but that is used only by tests.

We could have the setter be private and then try to get it with the proper reflection binding flags but there might still be issues.

Possible issues of adding the setter (internal or private) and/or using it by import :
1. Someone might use the setter without considering the implications
2. Using it in the import logic (without checking validity) might bring in bad paths.

Reply:
Those expected validations per each Location are handling when the settings is loaded here https://github.com/DynamoDS/Dynamo/blob/63e5c2f78f92a0144da7ae63fac4c855b3a28f1b/src/DynamoCore/Configuration/PreferenceSettings.cs#L368  by the helper https://github.com/DynamoDS/Dynamo/blob/63e5c2f78f92a0144da7ae63fac4c855b3a28f1b/src/DynamoUtilities/PathHelper.cs#L325